airflow/models/dagrun.py

```diff
@@ -263,10 +294,14 @@ def update_state(self, session=None):
         Determines the overall state of the DagRun based on the state
         of its TaskInstances.
 
-        :return: State
+        :return: state, schedulable_task_instances
+        :rtype: (State, list[TaskInstance])
         """
+        from airflow.ti_deps.deps.ready_to_reschedule import ReadyToRescheduleDep
+        from airflow.ti_deps.deps.not_in_retry_period_dep import NotInRetryPeriodDep
 
         dag = self.get_dag()
+        tis_to_schedule = []
 
         tis = self.get_task_instances(session=session)
         self.log.debug("Updating state for %s considering %s task(s)", self, len(tis))
```

Comment:
Do you think it is worth dividing the loop from line 272 into two loops? One loop will filters the elements and the second loop will set tasks on task instances. This does not affect performance, but will make it easier to understand the code.